### DISCUSSION TITLE
When using the API, the blocking mode is not working

### DISCUSSION THREAD

**Discussion by pprp** [langgenius/dify#8599] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I was trying to use API and it worked well using the following code:

```
curl -X POST 'https://api.dify.ai/v1/workflows/run' \
--header 'Authorization: Bearer app-xxxx' \
--header 'Content-Type: application/json' \
--data-raw '{
    "inputs": {"Target_language": "Chinese", "Input_language": "English", "Type": "xxx", "URL": "https://x1"},
    "response_mode": "blocking",xxxx
    "user": "abc-123"
}'
error code: 524%   
```

We got error code 524. I have tried using the same code and it worked fine.

**Comment by jpaautomatiza** [comment]
Create a node CODE and insert this code aften NODE HTTPS REQUEST.

Code:

// Recebe a entrada de dados dinâmicos do N8N

const items = $input.all(); // Captura todos os dados de entrada no N8N

let finalAnswer = "";

let conversationId = "";

let userId = ""; // Se precisar de um user_id, ajuste conforme o campo disponível

// Função para extrair o JSON de dentro da string 'data'

function extractJSONFromDataString(dataString) {

  // Extrai apenas o conteúdo JSON da string 'data: {...}'

  const jsonString = dataString.replace("data: ", "").trim();

  return JSON.parse(jsonString);

}

// Itera sobre os dados de entrada e processa cada um

for (let i = 0; i < items.length; i++) {

  // Verifica se a propriedade 'data' está presente e é uma string

  if (items[i].json.data && typeof items[i].json.data === 'string') {

    const dataLines = items[i].json.data.split("\n"); // Divide cada linha de "data" como string separada

    

    // Itera sobre cada linha de JSON em 'data'

    for (let j = 0; j < dataLines.length; j++) {

      if (dataLines[j].trim()) { // Ignora linhas vazias

        const parsedData = extractJSONFromDataString(dataLines[j]); // Extrai o JSON

        // Captura o conversation_id

        if (!conversationId && parsedData.conversation_id) {

          conversationId = parsedData.conversation_id;

        }

        // Se o evento for 'agent_message', concatenar a resposta

        if (parsedData.event === "agent_message" && parsedData.answer) {

          finalAnswer += parsedData.answer + " "; // Adiciona espaço entre as respostas

        }

        // Caso o evento seja 'agent_thought' e o campo 'thought' tenha valor, usa como resposta final

        if (parsedData.event === "agent_thought" && parsedData.thought) {

          finalAnswer = parsedData.thought; // Substitui a resposta final com o pensamento completo do agente

        }

      }

    }

  } else {

    // Loga um erro ou mensagem se a propriedade 'data' não existir ou não for uma string

    console.error(`Item ${i} não contém uma propriedade 'data' válida`);

  }

}

// Simula os dados adicionais para retriever_resources (baseado na sua saída de exemplo)

const retrieverResources = [

  {

    position: 1,

    dataset_id: "101b4c97-fc2e-463c-90b1-5261a4cdcafb",

    dataset_name: "iPhone",

    document_id: "8dd1ad74-0b5f-4175-b735-7d98bbbb4e00",

    document_name: "iPhone List",

    segment_id: "ed599c7f-2766-4294-9d1d-e5235a61270a",

    score: 0.98457545,

    content: `"Model","Release Date","Display Size","Resolution","Processor","RAM","Storage","Camera","Battery","Operating System"\n"iPhone 13 Pro Max","September 24, 2021","6.7 inch","1284 x 2778","Hexa-core (2x3.23 GHz Avalanche + 4x1.82 GHz Blizzard)","6 GB","128, 256, 512 GB, 1TB","12 MP","4352 mAh","iOS 15"`

  }

];

// Cria a saída final organizada

const finalOutput = {

  event: "message",

  message_id: "9da23599-e713-473b-982c-4328d4f5c78a", // Ajuste conforme necessário para gerar IDs dinâmicos

  conversation_id: conversationId,

  mode: "chat", // Presume-se que o modo seja 'chat'

  answer: finalAnswer.trim(), // Resposta final concatenada e sem espaços extras

  metadata: {

    usage: {

      prompt_tokens: 1033,

      prompt_unit_price: "0.001",

      prompt_price_unit: "0.001",

      prompt_price: "0.0010330",

      completion_tokens: 128,

      completion_unit_price: "0.002",

      completion_price_unit: "0.001",

      completion_price: "0.0002560",

      total_tokens: 1161,

      total_price: "0.0012890",

      currency: "USD",

      latency: 0.7682376249867957 // Simula um tempo de latência

    },

    retriever_resources: retrieverResources // Inclui os recursos simulados

  },

  created_at: Math.floor(Date.now() / 1000) // Timestamp atual em segundos

};

// Retorna a saída final

return finalOutput;
